src/bot.rs

```diff
@@ -59,151 +60,283 @@ impl Bot {
         }
     }
 
-    /// Creates a new request and adds a JSON message to it. The returned Future contains a the
-    /// reply as a string.  This method should be used if no file is added because a JSON msg is
-    /// always compacter than a formdata one.
-    pub fn fetch_json<'a>(
+    pub fn fetch_json(
         &self,
-        func: &str,
-        msg: &str,
-    ) -> impl Future<Item = String, Error = Error> + 'a {
-        debug!("Send JSON: {}", msg);
-
-        let mut header = List::new();
-        header.append("Content-Type: application/json").unwrap();
-
-        let mut a = Easy::new();
-        a.http_headers(header).unwrap();
-        a.post_fields_copy(msg.as_bytes()).unwrap();
-        a.post(true).unwrap();
-
-        self._fetch(func, a)
+        func: String,
+        msg: String,
+    ) -> impl Future<Item = String, Error = Error> {
+        fetch_json(self.session.clone(), self.key.clone(), func, msg)
     }
 
-    /// Creates a new request with some byte content (e.g. a file). The method properties have to be
-    /// in the formdata setup and cannot be sent as JSON.
-    pub fn fetch_formdata<'a, T>(
+    pub fn fetch_formdata<T>(
         &self,
-        func: &str,
+        func: String,
         msg: Value,
-        mut file: T,
+        file: T,
         kind: &str,
         file_name: &str,
-    ) -> impl Future<Item = String, Error = Error> + 'a
+    ) -> impl Future<Item = String, Error = Error>
     where
         T: io::Read,
     {
-        debug!("Send formdata: {}", msg.to_string());
-        let mut content = Vec::new();
+        fetch_formdata(
```

Comment:
No need for a seperate function, which is static, both could just live in a single function

Reply:
The functions have been returned to their original place inside the Bot impl